### ISSUE TITLE
Missing id in hcloud_image data source

### ISSUE THREAD

**Issue by codello** [hetznercloud/terraform-provider-hcloud#306]
The `hcloud_image` data source allows to dynamically select an image for a server. However when the data source depends on another resource (and thus is not executed when creating a plan) it is missing its `id` attribute which can cause issues when creating servers that depend on an image from a `hcloud_image` data source.

I think this is a bug in the provider. Or is this expected behavior? How does one dynamically find an image using the `hcloud` provider when the name of the image (or the selector) depends on another resource?

### Terraform Version
```
Terraform v0.14.5
+ provider registry.terraform.io/hashicorp/time v0.6.0
+ provider registry.terraform.io/hetznercloud/hcloud v1.24.0
```

### Terraform Configuration Files
```terraform
terraform {
  required_version = ">= 0.14"

  required_providers {
    hcloud = {
      source  = "hetznercloud/hcloud"
      version = "~> 1.24.0"
    }
  }
}

provider "hcloud" {
  token = "..."
}

resource "time_sleep" "wait" {
  create_duration = "1s"
  triggers = {
    name = "centos-8"
  }
}

data "hcloud_image" "image" {
  name = time_sleep.wait.triggers.name
  with_status   = ["available"]
}

resource "hcloud_server" "server" {
  name        = "test"
  server_type = "cx11"
  location    = "nbg1"
  image       = data.hcloud_image.image.id
}
```
**Note**: I'm aware that this specific example could be resolved by using `depends_on = [time_sleep.wait]` in the data source declaration and then specifying `name = "centos-8"` directly.

### Expected Behavior
When running `terraform plan` I expect a plan to be created that creates a single server (and the `wait` resource).

### Actual Behavior
The plan fails with the following error.
```
Error: Required attribute is not set

  on main.tf line 32, in resource "hcloud_server" "server":
  32:   image       = data.hcloud_image.image.id
```

### Steps to Reproduce
1. Create a file with the contents under `Terraform Configuration Files`. Substitute the token for the `hcloud` provider.
2. `terraform init`
3. `terraform plan`

### Additional Context
It seems to me that this issue is only related to the schema of the `hcloud_image` data source. When the data source is actually run (and it finds an image) the `id` attribute is available. So the following **does** create the resources as expected:
1. Run `terraform apply -target=time_sleep.wait`
2. Run `terraform apply`

The schema of the data source specifies the `id` as `Optional: true`. A fix for this issue might be as simple as adding `Computed: true` (this is just a guess, I have no experience with the schema specifications).
https://github.com/hetznercloud/terraform-provider-hcloud/blob/0e17403f4109d38460d07a1b7a46e6b9dc244634/internal/image/data_source.go#L23-L26

**Comment by LKaemmerling** [comment]
Hey @codello,

thank you! I was able to reproduce the problem and also have a fixed version locally already running. I will prepare a MR with a fix. Thank you!

- Lukas

**Comment by LKaemmerling** [comment]
Hey @codello,

we released v1.24.1 where this bug was fixed. You can update by changing the version in your terraform provider configuration.

**Comment by codello** [comment]
Works perfectly. Thanks a lot!
